lib/teracy-dev-core.rb

```diff
@@ -26,10 +25,8 @@ def self.init
     TeracyDev.register_configurator(TeracyDevCore::Config::SSH.new)
     TeracyDev.register_configurator(TeracyDevCore::Config::SyncedFolders.new)
     TeracyDev.register_configurator(TeracyDevCore::Config::Vgrant.new)
-    TeracyDev.register_configurator(TeracyDevCore::Config::VirtualBoxProvider.new)
-    TeracyDev.register_configurator(TeracyDevCore::Config::VMwareProvider.new) # Use for vmware provider.
+    TeracyDev.register_configurator(TeracyDevCore::Config::Provider.new)
     TeracyDev.register_configurator(TeracyDevCore::Config::WinRM.new)
     TeracyDev.register_configurator(TeracyDevCore::Config::WinSSH.new)
   end
```

Comment:
Let's add public api for the TeracyDevCore:

TeracyDevCore.register_provider(provider_type: String, provider: Provider)

TeracyDevCore.register_provider is the proxy to the TeracyDevCore::Providers::Manager

Providers::Manager is a static class that holds the registered providers (registry = {"x_type": [x_provider_impl]})
Providers::Manager allows other classes to access the registry

Reply:
@hieptranquoc check my updated comment here